bin/check-aggregate-output.rb

```diff
@@ -0,0 +1,209 @@
+         short: '-k',
+         boolean: true,
+         description: 'Enabling insecure connections',
+         default: false
+
+  option :user,
+         short: '-u USER',
+         long: '--user USER',
+         description: 'Sensu API USER'
+
+  option :password,
+         short: '-p PASSWORD',
+         long: '--password PASSWORD',
+         description: 'Sensu API PASSWORD'
+
+  option :timeout,
+         short: '-t SECONDS',
+         long: '--timeout SECONDS',
+         description: 'Sensu API connection timeout in SECONDS',
+         proc: proc(&:to_i),
+         default: 30
+
+  option :check,
+         short: '-c CHECK',
+         long: '--check CHECK',
+         description: 'Aggregate CHECK name',
+         required: true
+
+  option :age,
+         short: '-A SECONDS',
+         long: '--age SECONDS',
+         description: 'Minimum aggregate age in SECONDS, time since check request issued',
+         default: 30,
+         proc: proc(&:to_i)
+
+  option :message,
+         short: '-M MESSAGE',
+         long: '--message MESSAGE',
+         description: 'A custom error MESSAGE'
+
+  option :collect_output,
+         short: '-o',
+         long: '--output',
+         boolean: true,
+         description: 'Collects all non-ok outputs',
+         default: true
+
+  option :warning,
+         short: '-W PERCENT',
+         long: '--warning PERCENT',
+         description: 'PERCENT warning before warning (can be change with --ignore-severity)',
+         proc: proc(&:to_i)
+
+  option :warning_count,
+         long: '--warning_count INTEGER',
+         description: 'number of nodes in warning before warning (can be change with --ignore-severity)',
+         proc: proc(&:to_i)
+
+  option :critical,
+         short: '-C PERCENT',
+         long: '--critical PERCENT',
+         description: 'PERCENT critical before critical (can be change with --ignore-severity)',
+         proc: proc(&:to_i)
+
+  option :critical_count,
+         long: '--critical_count INTEGER',
+         description: 'number of node in critical before critical (can be change with --ignore-severity)',
+         proc: proc(&:to_i)
+
+  option :ignore_severity,
+         long: '--ignore-severity',
+         description: 'Ignore severities, all non-ok will count for critical, critical_count, warning and warning_count option',
+         boolean: true,
+         default: false
+
+  def api_request(resource)
+    verify_mode = OpenSSL::SSL::VERIFY_PEER
+    verify_mode = OpenSSL::SSL::VERIFY_NONE if config[:insecure]
+    request = RestClient::Resource.new(config[:api] + resource, timeout: config[:timeout],
+                                                                user: config[:user],
+                                                                password: config[:password],
+                                                                verify_ssl: verify_mode)
+    JSON.parse(request.get)
+  rescue Errno::ECONNREFUSED
+    warning 'Connection refused'
```

Comment:
You don't need to change anything here but in general it is better to return a status and a message from a function rather than executing immediately as it makes it harder to write tests for small units of code.